src/Stratis.Features.FederatedPeg/CollateralFederationManager.cs

```diff
@@ -36,6 +36,19 @@ public override void Initialize()
             }
         }
 
+        protected override void AddFederationMemberLocked(IFederationMember federationMember)
+        {
+            var collateralMember = federationMember as CollateralFederationMember;
+
+            if (this.federationMembers.Cast<CollateralFederationMember>().Any(x => x.CollateralMainchainAddress == collateralMember.CollateralMainchainAddress))
```

Comment:
A possibility exists that `collateralMember` can be null here, so you need to protect against it.

Reply:
It can't. All fed members are collateral members in case collateral fed manager is enabled.